tests/test_io.py

```diff
@@ -1,10 +1,26 @@
 import pkgutil
+import gzip
 
-from nose.tools import eq_
+from nose.tools import eq_, assert_raises, assert_false, assert_true
 
 from unittest import TestCase
 
-from edgePy.io import *  # Test import of __all__
+from edgePy.io import DataImporter, GroupImporter, get_dataset_path, get_open_function  # Test import of __all__
+
+
+class TestCommonFunctions(TestCase):
+
+    def test_get_open_function(self):
+
+        file_name = "blah.gz"
+        open_function, decode_required = get_open_function(filename=file_name)
+        eq_(gzip.open, open_function)
+        assert_true(decode_required)
+
+        file_name = "blah.txt"
+        open_function, decode_required = get_open_function(filename=file_name)
+        eq_(open, open_function)
```

Comment:
Should not test functions for equality. This is actually a tricky problem in unit testing. Because Python allows for duck-typing (as you are doing here) it is better to test the functions do the right thing.

You should read a file with the proper format with all open functions and confirm that after reading the lines are equal. You can do this with unittest's `self.assertMultiLineEqual`.

Reply:
The file you make for testing should be virtual or temprorary and not physical to prevent clutter.